src/resolvers/Mutation/updatePost.ts

```diff
@@ -0,0 +1,71 @@
+import { MutationResolvers } from "../../types/generatedGraphQLTypes";
+import { errors, requestContext } from "../../libraries";
+import { User, Post } from "../../models";
+import {
+  USER_NOT_FOUND_MESSAGE,
+  USER_NOT_FOUND_CODE,
+  USER_NOT_FOUND_PARAM,
+  USER_NOT_AUTHORIZED_MESSAGE,
+  USER_NOT_AUTHORIZED_CODE,
+  USER_NOT_AUTHORIZED_PARAM,
+  POST_NOT_FOUND_MESSAGE,
+  POST_NOT_FOUND_CODE,
+  POST_NOT_FOUND_PARAM,
+} from "../../constants";
+
+export const updatePost: MutationResolvers["updatePost"] = async (
+  _parent,
+  args,
+  context
+) => {
+  const currentUserExists = await User.exists({
+    _id: context.userId,
+  });
+
+  // checks if current user exists
+  if (currentUserExists === false) {
+    throw new errors.NotFoundError(
+      requestContext.translate(USER_NOT_FOUND_MESSAGE),
+      USER_NOT_FOUND_CODE,
+      USER_NOT_FOUND_PARAM
+    );
+  }
+
+  const post = await Post.findOne({
+    _id: args.id,
+  }).lean();
+
+  // checks if there exists a post with _id === args.id
+  if (!post) {
+    throw new errors.NotFoundError(
+      requestContext.translate(POST_NOT_FOUND_MESSAGE),
+      POST_NOT_FOUND_CODE,
+      POST_NOT_FOUND_PARAM
+    );
+  }
+
+  const currentUserIsPostAdmin =
+    post.creator.toString() === context.userId.toString();
+
```

Comment:
Post only has a creator not admin. Name it properly to not confuse people.

Reply:
Done.